package_linter.py

```diff
@@ -354,6 +359,52 @@ def check_arg_retrieval(script):
         print_wrong("Argument retrieval from manifest with $1 is deprecated. You may use $YNH_APP_ARG_*")
         print_wrong("For more details see: https://yunohost.org/#/packaging_apps_arguments_management_en")
 
+def check_helper_usage_dependencies(script_name):
+    """
+    Detect usage of ynh_package_* & apt-get *
+    and suggest herlpers ynh_install_app_dependencies and ynh_remove_app_dependencies
+    """
+    script = open(script_name).read()
+
+    if "ynh_package_install" in script or "apt-get install" in script:
+        print_warning("You should not use `ynh_package_install` or `apt-get install`, use `ynh_install_app_dependencies` instead")
+
+    if "ynh_package_remove" in script or "apt-get remove" in script:
+        print_warning("You should not use `ynh_package_remove` or `apt-get removeè, use `ynh_remove_app_dependencies` instead")
+
+def check_helper_usage_unix(script_name):
+    """
+    Detect usage of unix commands with helper equivalents:
+    - sudo    → nothing
+    - rm      → ynh_secure_remove
+    - sed -i  → ynh_replace_string
+    """
+    script = open(script_name).read()
+
+    if "rm -rf" in script:
+        print_warning("You should not use `rm -rf`, use ynh_secure_remove instead")
+
+    if "sed -i" in script:
+        print_warning("You should not have to use `sed -i`, use ynh_replace_string or one of the helper")
+
+    if "sudo " in script:
+        print_warning("You should not have to use `sudo`, the script is run as root")
```

Comment:
sudo can be used to run a command as another user... I am not sure this check is relevant.

Reply:
`su -l user -c "cmd"` or a helper might be created for this kind of actions.
It thought [it was already implemented](https://helpers.yunohost.org/), but I didn’t manage to find it.

Reply:
https://github.com/YunoHost-Apps/Experimental_helpers/blob/master/ynh_exec_as/ynh_exec_as